ramips: add support for TP-Link RE220 v2

rowanborder opened the pull request:
TP-Link RE220 v2 is a wireless range extender with Ethernet and 2.4G and 5G
WiFi with internal antennas. It's based on MediaTek MT7628AN+MT7610EN.

This port of OpenWRT leverages work done by Andreas Böhler <dev@aboehler.at>
for the TP-Link RE200 v2 as both devices share the same SoC, flash layout
and GPIO pinout.

Specifications

MediaTek MT7628AN (580 Mhz)
64 MB of RAM
8 MB of FLASH
2T2R 2.4 GHz and 1T1R 5 GHz
1x 10/100 Mbps Ethernet
UART header on PCB (57600 8n1)
8x LED (GPIO-controlled), 2x button
There are 2.4G and 5G LEDs in red and green which are controlled separately.

Web Interface Installation

It is possible to upgrade to OpenWrt via the web interface. Simply flash
the -factory.bin from OEM. In contrast to a stock firmware, this will not
overwrite U-Boot.

Signed-off-by: Rowan Border <rowanjborder@gmail.com>

adrianschmutzler commented:
Please add installation/flashing instructions to your commit message.

adrianschmutzler commented on target/linux/ramips/dts/mt7628an_tplink_re220-v2.dts:
```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7628an_tplink_re220.dtsi"
```
If this really is a 100 % clone of the RE200 v2, and you didn't just copy everything without testing, then you might use the existing DTSI from the re200 here, as that's what the DTSI files are for (and drop your 1:1 DTSI copy).

rowanborder replied:
Yes, both devices share the same flash layout and GPIO pinout.

adrianschmutzler commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -345,6 +345,16 @@ define Device/tplink_archer-c50-v4
 endef
 TARGET_DEVICES += tplink_archer-c50-v4
 
+define Device/tplink_re220-v2
```
Please move to have proper sorting.

rowanborder replied:
Done.

adrianschmutzler commented:
This is just a little too much C+P for me. Did you test everything, and have you checked whether really all of the support strings and partitions for tplink-safeloader are really the same?

rowanborder committed:
ramips: add support for TP-Link RE220 v2

TP-Link RE220 v2 is a wireless range extender with Ethernet and 2.4G and 5G
WiFi with internal antennas. It's based on MediaTek MT7628AN+MT7610EN.

This port of OpenWRT leverages work done by Andreas Böhler <dev@aboehler.at>
for the TP-Link RE200 v2 as both devices share the same SoC, flash layout
and GPIO pinout.

Specifications

MediaTek MT7628AN (580 Mhz)
64 MB of RAM
8 MB of FLASH
2T2R 2.4 GHz and 1T1R 5 GHz
1x 10/100 Mbps Ethernet
UART header on PCB (57600 8n1)
8x LED (GPIO-controlled), 2x button
There are 2.4G and 5G LEDs in red and green which are controlled separately.

Web Interface Installation

It is possible to upgrade to OpenWrt via the web interface. Simply flash
the -factory.bin from OEM. In contrast to a stock firmware, this will not
overwrite U-Boot.

Signed-off-by: Rowan Border <rowanjborder@gmail.com>

rowanborder commented:
> Please add installation/flashing instructions to your commit message.

Done.

adrianschmutzler commented:
Commit title: rampis -> ramips

rowanborder commented:
> This is just a little too much C+P for me. Did you test everything, and have you checked whether really all of the support strings and partitions for tplink-safeloader are really the same?

Yes, I have successfully deployed this OpenWRT port on three different RE220 devices.

rowanborder commented:
> Commit title: rampis -> ramips



> Commit title: rampis -> ramips

Done.

adrianschmutzler commented:
Thanks, merged.
